cpp/include/cudf/detail/utilities/rmm_host_vector.hpp

```diff
@@ -0,0 +1,232 @@
+
+  /**
+   * @brief This method returns the address of a \c reference of
+   *  interest.
+   *
+   *  @param r The \c reference of interest.
+   *  @return \c r's address.
+   */
+  __host__ __device__ inline pointer address(reference r) { return &r; }
+
+  /**
+   * @brief This method returns the address of a \c const_reference
+   *  of interest.
+   *
+   *  @param r The \c const_reference of interest.
+   *  @return \c r's address.
+   */
+  __host__ __device__ inline const_pointer address(const_reference r) { return &r; }
+
+  /**
+   * @brief This method allocates storage for objects in host memory.
+   *
+   *  @param cnt The number of objects to allocate.
+   *  @return a \c pointer to the newly allocated objects.
+   *  @note The second parameter to this function is meant as a
+   *        hint pointer to a nearby memory location, but is
+   *        not used by this allocator.
+   *  @note This method does not invoke \p value_type's constructor.
+   *        It is the responsibility of the caller to initialize the
+   *        objects at the returned \c pointer.
+   */
+  __host__ inline pointer allocate(size_type cnt, const_pointer /*hint*/ = 0)
+  {
+    if (cnt > this->max_size()) { throw std::bad_alloc(); }  // end if
+    return static_cast<pointer>(mr.allocate(cnt * sizeof(value_type)));
+  }
+
+  /**
+   * @brief This method deallocates host memory previously allocated
+   *  with this \c rmm_host_allocator.
+   *
+   *  @param p A \c pointer to the previously allocated memory.
+   *  @note The second parameter is the number of objects previously allocated.
+   *  @note This method does not invoke \p value_type's destructor.
+   *        It is the responsibility of the caller to destroy
+   *        the objects stored at \p p.
+   */
+  __host__ inline void deallocate(pointer p, size_type cnt)
+  {
+    mr.deallocate(p, cnt * sizeof(value_type));
+  }
+
+  /**
+   * @brief This method returns the maximum size of the \c cnt parameter
+   *  accepted by the \p allocate() method.
+   *
+   *  @return The maximum number of objects that may be allocated
+   *          by a single call to \p allocate().
+   */
+  inline size_type max_size() const { return (std::numeric_limits<size_type>::max)() / sizeof(T); }
+
+  /**
+   * @brief This method tests this \p rmm_host_allocator for equality to
+   *  another.
+   *
+   *  @param x The other \p rmm_host_allocator of interest.
+   *  @return This method always returns \c true.
+   */
+  __host__ __device__ inline bool operator==(rmm_host_allocator const& x) const
+  {
+    return x.mr == mr;
+  }
+
+  /**
+   * @brief This method tests this \p rmm_host_allocator for inequality
+   *  to another.
+   *
+   *  @param x The other \p rmm_host_allocator of interest.
+   *  @return This method always returns \c false.
+   */
+  __host__ __device__ inline bool operator!=(rmm_host_allocator const& x) const
+  {
+    return !operator==(x);
+  }
+
+ private:
+  cudf::host_resource_ref mr;
```

Comment:
I'm surprised you don't want stream-ordered allocation for your pinned memory. Without it the `pool_memory_resource` will do everything on the default stream.